### PR TITLE
Add tests via TestDriverAI

### PR DESCRIPTION
**Pull request by chottuthejimmy**
# Description
This PR contains the test cases to test the basic functionality of the [stirling.io](https://stirlingpdf.io/) webpage.
The tests are written using [testdriverai](https://testdriver.ai) a next gen automation AI agent for end-to-end testing of web and desktop application.

## Overview

This PR adds TestDriver tests that I generated using the `testdriverai` npm package. Instead of using selectors, TestDriver uses prompts and AI vision to complete tasks. This means the tests will still work even when code and design changes.

Another huge benefit is we can test PDF input and output because tests operate at the OS level.
You can [see the schema of supported commands here](https://docs.testdriver.ai/reference/yml-schema).
## Test Cases

- Opens Google Chrome and interacts with various UI elements like buttons and icons.
- Verifies opening and viewing of a PDF file.
- Navigates back to the homepage after certain actions.
- Changes the language from English to Spanish and back, verifying text translations.
- Toggles the website theme using sun and moon icons.
- Searches for specific terms and verifies correct navigation to related pages.
- Hovers over buttons like "Merge" and checks for visual feedback (background changes).

## Setup

In order to complete this PR you’ll need to add a API key as a GitHub action secret. [Follow the directions here for more info.](https://docs.testdriver.ai/continuous-integration/github-setup)
Once you sign up, let us know your email and we’ll enroll you in the open source program for free Pro account.

## Here's how to add more tests :

- Install the testdriverai package via npm and start by creating tests by just prompting the cli using natural language prompts ( you can check the `prompts` field in the test file for motivation )
- Once you are done writing the tests, just push it to the repo, here's where the magic happens.
- Whenever there's a new PR, the VMs (maintained by us) will clone the feature branch and performs the tests and returns the result along with the screen recordings and its logs (thanks to [dashcam.io](https://dashcam.io/))
- Also feel free to check out our docs on [how to improve performance](https://docs.testdriver.ai/continuous-integration/performance) and techniques like [parallel testing](https://docs.testdriver.ai/continuous-integration/parallel-testing)

## What’s next

We can set this up to build feature branches using our [prerun scripts](https://docs.testdriver.ai/continuous-integration/prerun-scripts).

## Checklist

- [x] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have read the section [Add New Translation Tags](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/HowToAddNewLanguage.md#add-new-translation-tags) (for new translation tags only)

### CODE REVIEW

File: testdriver/test.yml
Comment (Frooodle) [inline testdriver/test.yml]: did it not work? :')

**Comment by ianjennings** [comment]
@Frooodle We need your approval to run the workflow.

CleanShot 2024-09-24 at 12 28 52

**Comment by Frooodle** [comment]
I believe I need the 
          key: ${{secrets.TESTDRIVER_API_KEY}}

As well? 
Additionally we should set it up to run on PR code not on web instance :p

File: .github/workflows/testdriver.yml
Comment (Frooodle) [inline .github/workflows/testdriver.yml:35]: Why does it need GitHub token here of it doesn't connect with github

**Review by Frooodle — CHANGES_REQUESTED** [review]

File: .github/workflows/testdriver.yml
Comment (ianjennings) [reply .github/workflows/testdriver.yml:35]: > Why does it need GitHub token here of it doesn't connect with github

In order to run PR code, it clones the repo down onto the VM.

We then set up the repo using the `prerun` script
https://docs.testdriver.ai/continuous-integration/prerun-scripts

**Comment by ianjennings** [comment]
Here is a preview of the test execution:

[testdriverai/Stirling-PDF/refs/heads/main](https://app.dashcam.io/replay/66f2d3b88a4be78943aeb893?share=7n8vDeLihQj4aCHuPwOKWA)

Watch [testdriverai/Stirling-PDF/refs/heads/main](https://app.dashcam.io/replay/66f2d3b88a4be78943aeb893?share=7n8vDeLihQj4aCHuPwOKWA) on Dashcam

File: testdriver/test.yml
Comment (chottuthejimmy) [reply testdriver/test.yml]: ya so the issue was it clicked on the search bar in the taskbar a couple of times, so we felt it would be better to remove this and make sure that it always does the correct stuff .
So we are currently working on to spin up the VMs with auto-hide taskbar by default and will add this back once thats done.

File: .github/workflows/testdriver.yml
Comment (ianjennings) [reply .github/workflows/testdriver.yml:35]: Oh sorry, actually since the repo is OSS the GH_TOKEN is optional. It will use our 'testdriverbot' token instead.

**Comment by chottuthejimmy** [comment]
Hey @Frooodle could you have another look at this.

It's a go from our side!

Cheers!

**Comment by Frooodle** [comment]
Sorry been very busy should be able to properly check it end of week

**Comment by Frooodle** [comment]
Added key but job still failed, does account need upgraded?

**Comment by chottuthejimmy** [comment]
Hey @Frooodle could please make sure you have got the API set up properly. 
I would be happy to hop on a call and help you out, so let me know.

**Comment by Frooodle** [comment]
Secret is there along side other secrets used within this repo,
image

**Comment by chottuthejimmy** [comment]
Hey @Frooodle, so I tried to recreate the issue by forking and making a PR. 
I think its pretty much the API issue and nothing more. So could you try out a couple of things for us please. 

Go to [dashcam.io/team](https://app.dashcam.io/team) and click on the cycle button next to `API Key`, copy it and click on update. And replace the secret with this new key. 
image

Try to re-run the failed job over [here](https://github.com/Stirling-Tools/Stirling-PDF/actions/runs/11111172691)

If that doesn't work, make sure the right permissions are set [here](https://github.com/Stirling-Tools/Stirling-PDF/settings/actions) for PRs to access the secrets, or you can just merge this PR if you do not wish to change the permission.

**Comment by Frooodle** [comment]
All seems okay my side from what I can see,
Can we add a 
      - name: Debug - Print masked secret
        run: |
          echo "Secret is: ${{ secrets.TESTDRIVER_API_KEY }}"
          
          to the job just to see if it can grab it or if its being lost in the call to the github action

**Comment by Frooodle** [comment]
Hmm printing nope
I'll check more my side

**Comment by chottuthejimmy** [comment]
Is it because of this [code](https://github.com/Stirling-Tools/Stirling-PDF/blob/97734efb712a6870226d621cf8a5dd12fb0100ec/.github/workflows/testdriver.yml#L3-L7) ??

**Comment by chottuthejimmy** [comment]
@Frooodle could you take a look at this sometime today please!

**Comment by Frooodle** [comment]
Merging into repo branch to test it

**Comment by Frooodle** [comment]
and that instantly worked.. looks like it had to be integrated first to have access to secret, it works fine now

**Comment by Frooodle** [comment]
going to monitor PR and see ETA runtimes etc, will need to have it run a docker container and test the actual code not stirlingpdf.io deployment before merging into main

**Comment by Frooodle** [comment]
https://app.dashcam.io/replay/6706e85e67f936f4742aef4d?share=3NOziidOgB0CDw1kNjVaA

Did show some package.json error

**Comment by chottuthejimmy** [comment]
> and that instantly worked.. looks like it had to be integrated first to have access to secret, it works fine now

Told you [hear](https://github.com/Stirling-Tools/Stirling-PDF/pull/1957#issuecomment-2388515397) 
JK.

**Comment by chottuthejimmy** [comment]
> [](https://app.dashcam.io/replay/6706e85e67f936f4742aef4d?share=3NOziidOgB0CDw1kNjVaA)
> 
> Did show some package.json error

We are unable to access the recordings (as they belong to your org / acc) please accept my request access so that we can have a look into it

**Pull request merged**
